src/cluster.c

```diff
@@ -1364,6 +1364,36 @@ unsigned int keyHashSlot(char *key, int keylen) {
     return crc16(key+s+1,e-s-1) & 0x3FFF;
 }
 
+/* If it can be inferred that the given glob-style pattern, as implemented in
+ * stringmatchlen() in util.c, only can match keys belonging to a single slot,
+ * that slot is returned. Otherwise -1 is returned. */
+int patternHashSlot(char *pattern, int length) {
+    int s = -1; /* index of the first '{' */
+
+    for (int i = 0; i < length; i++) {
+        if (pattern[i] == '*' || pattern[i] == '?' || pattern[i] == '[') {
+            /* Wildcard or character class found. Keys can be in any slot. */
+            return -1;
+        } else if (pattern[i] == '\\') {
```

Comment:
Why return -1 when '\\' is found? I tried and it seems Redis is capable of computing slots of keys with '\\'.

Reply:
The explanation is in the comment on the next two lines:

```C
            /* Escaped character. Computing slot in this case is not
             * implemented. We would need a temp buffer. */
```

The glob-like syntax has backslash escapes. Before we compute the slot, we would need to remove the backslash. See util.c for the full syntax of patterns.

Reply:
I see, it turns out that `scan 0 match a\n` can also match `an`, which is in a different slot from `a\n`...